linkis-computation-governance/linkis-manager/linkis-application-manager/src/main/java/org/apache/linkis/manager/rm/service/RequestResourceService.java

```diff
@@ -404,7 +403,61 @@ public Pair<Integer, String> generateNotEnoughMessage(
         return generateNotEnoughMessage(
             dy.getYarnResource(), dyAvailable.getYarnResource(), dyMax.getYarnResource());
       }
-
+    } else if (requestResource instanceof KubernetesResource) {
+      KubernetesResource kubernetesResource = (KubernetesResource) requestResource;
+      KubernetesResource kubernetesResourceAvailable = (KubernetesResource) availableResource;
+      KubernetesResource kubernetesResourceMax = (KubernetesResource) maxResource;
+      if (kubernetesResource.getCores() > kubernetesResourceAvailable.getCores()) {
+        return Pair.of(
+            RMErrorCode.NAMESPACE_CPU_INSUFFICIENT.getErrorCode(),
+            RMErrorCode.NAMESPACE_CPU_INSUFFICIENT.getErrorDesc()
+                + RMUtils.getResourceInfoMsg(
+                    RMConstant.CPU,
+                    RMConstant.KUBERNETES_CPU_UNIT,
+                    kubernetesResource.getCores(),
+                    kubernetesResourceAvailable.getCores(),
+                    kubernetesResourceMax.getCores()));
+      } else if (kubernetesResource.getMemory() > kubernetesResource.getMemory()) {
```

Comment:
change to "kubernetesResource.getMemory() > kubernetesResourceAvailable.getMemory()"